src/tests/unit/components/update-pane/update-pane-test.tsx

```diff
@@ -121,10 +125,17 @@ describe('test update pane', () => {
     const { container } = wrappedRender(<UpdatePaneTestComponent />, { store });
 
     const updateAction = UMLElementRepository.updateStart(umlClass.id);
-    store.dispatch(updateAction);
+    act(() => {
+      store.dispatch(updateAction);
+    });
 
-    jest.runAllTimers();
-    fireEvent.pointerDown(container.querySelector('svg')!);
+    act(() => {
+      jest.runAllTimers();
+    });
+
+    act(() => {
+      fireEvent.pointerDown(container.querySelector('svg')!);
+    });
```

Comment:
You are breaking up `act(...)` statements into multiple ones, why not use one? As far as I can tell from the stuff I saw, it should be fine to wrap all the code in one `act` statement.

Reply:
Thanks for the suggestion Felix. Initially tried the approach you suggested but didn't work at all. Usually order is important in Apollon. We have to create the application and then use getter and setter functions to get/modify application state. Using single act probably breaks this logic.

Reply:
Yeah, you are right. The way you did it is probably safer 👍